maven-slf4j-wrapper/src/main/java/org/apache/maven/logwrapper/LogLevelRecorder.java

```diff
@@ -0,0 +1,55 @@
+package org.apache.maven.logwrapper;
+
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import org.slf4j.event.Level;
+
+/**
+ * Responsible for keeping state of whether the threshold of the --fail-level flag has been hit.
+ */
+public class LogLevelRecorder
+{
+    private final Level logThreshold;
+    private boolean thresholdHit = false;
+
+    public LogLevelRecorder( String threshold )
+    {
+        Level level = Level.valueOf( threshold );
+        if ( level.toInt() < Level.WARN.toInt() )
+        {
+            throw new IllegalArgumentException( "Logging level thresholds can only be set to WARN or ERROR" );
+        }
+
+        logThreshold = level;
+    }
+
+    public void record( Level logLevel )
+    {
+        if ( !thresholdHit && logLevel.toInt() >= logThreshold.toInt() )
+        {
+            thresholdHit = true;
+        }
+    }
+
+    public boolean isThresholdHit()
```

Comment:
I prefer 2-word based methodnames. How about `hitThreshold`? Or something else...

Reply:
`hitThreshold` could be confiused with a numeric value.

Reply:
`Contains` feels like it's a collection. What about `thresholdRecorded`?

Reply:
I'd like to start with the verb

Reply:
We could go for `reachedTreshold` or `onTreshold`?

Reply:
The latter sounds like a callback method on an event handler. I'd go with the former.

Reply:
I agree, what do you think @rfscholte ?

Reply:
After offline discussion with rfscholte and luc we went with `metThreshold`. What do you think @michael-o ?